comp/core/config/params.go

```diff
@@ -0,0 +1,145 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package config
+
+import "github.com/DataDog/datadog-agent/cmd/agent/common"
+
+// Params defines the parameters for the config component.
+type Params struct {
+	// confFilePath is the path at which to look for configuration, usually
+	// given by the --cfgpath command-line flag.
+	confFilePath string
+
+	// configName is the root of the name of the configuration file.  The
+	// comp/core/config component will search for a file with this name
+	// in ConfFilePath, using a variety of extensions.  The default is
+	// "datadog".
+	configName string
+
+	// sysProbeConfFilePath is the path at which to look for system-probe
+	// configuration, usually given by --sysprobecfgpath.  This is not used
+	// unless ConfigLoadSysProbe is true.
+	sysProbeConfFilePath string
+
+	// ConfigLoadSysProbe determines whether to read the system-probe.yaml into
+	// the component's config data.
+	configLoadSysProbe bool
+
+	// securityAgentConfigFilePaths are the paths at which to look for security-aegnt
+	// configuration, usually given by the --cfgpath command-line flag.
+	securityAgentConfigFilePaths []string
+
+	// configLoadSecurityAgent determines whether to read the config from
+	// SecurityAgentConfigFilePaths or from ConfFilePath.
+	configLoadSecurityAgent bool
+
+	// ConfigLoadSecrets determines whether secrets in the configuration file
+	// should be evaluated.  This is typically false for one-shot commands.
+	configLoadSecrets bool
+
+	// configMissingOK determines whether it is a fatal error if the config
+	// file does not exist.
+	configMissingOK bool
+
+	// defaultConfPath determines the default configuration path.
+	// if defaultConfPath is empty, then no default configuration path is used.
+	defaultConfPath string
+}
+
+// NewParams creates a new instance of Params
+func NewParams(defaultConfPath string, options ...func(*Params)) Params {
+	params := Params{
+		defaultConfPath: defaultConfPath,
+	}
+	for _, o := range options {
+		o(&params)
+	}
+	return params
+}
+
+// NewAgentParams creates a new instance of Params for the Agent.
+func NewAgentParams(confFilePath string, configLoadSecrets bool, options ...func(*Params)) Params {
```

Comment:
WDYT about having 2 methods for the agent `NewAgentParamsWithSecrets` and `NewAgentParamsWithoutSecret`. We're going to use this in a lot of places and a bool parameter is never very obvious (ex `NewAgentParams("", true)`). Or `NewAgentParams` and `NewAgentParamsWithSecrets` ?

Reply:
Code updated.